src/software/thunderscope/robot_diagnostics/error_log_widgets.py

```diff
@@ -0,0 +1,273 @@
+from pyqtgraph.Qt.QtWidgets import *
+from pyqtgraph.Qt.QtCore import Qt
+from pyqtgraph.Qt import QtGui
+from proto.import_all_protos import *
+from datetime import datetime
+import textwrap
+import software.thunderscope.robot_diagnostics.robot_error_log_icons.error_log_constants as error_constants
+from proto.robot_log_msg_pb2 import RobotLog, LogLevel
+from software.thunderscope.robot_diagnostics.robot_status import RobotStatusView
+from software.thunderscope.constants import LOG_LEVEL_STR_MAP
+
+
+class RobotLogMessageWidget(QFrame):
+    """
+    A single error message from a robot.
+
+    Displays an icon corresponding to the type of error, the robot id, the message
+    and the timestamp of the error
+    """
+
+    def __init__(
+        self, robot_id: int, message: str, icon: QtGui.QPixmap, timestamp=None
+    ):
+        super(RobotLogMessageWidget, self).__init__()
+
+        self.layout = QHBoxLayout()
+        self.icon = QLabel()
+
+        # setting up the icon
+        self.icon_size = self.width() / 10
```

Comment:
I think this widget can generally be made more compact so `RobotErrorLog` can be viewed as a smaller overall widget. During an actual match, we layout Thunderscope so we show almost every widget simultaneously on the screen, so it's important for the widgets to not take too much screen space.